modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/bean/APICreationRequestBean.java

```diff
@@ -122,6 +123,39 @@ public void setTechOwnerMail(String techOwnerMail) {
     /**
      * constructor of the APICreationRequestBean class only with production url
      *
+     * @param apiName       - Name of the APi
+     * @param context       - API context
+     * @param version       - API version
+     * @param provider      - Api Provider
+     * @param apiTier       - Api Tier
+     * @param resourceTier- Resource Tier
+     * @param endpointUrl   - Endpoint URL of the API
+     * @throws APIManagerIntegrationTestException - Exception throws when constructing the end point url JSON
+     */
+    public APICreationRequestBean(String apiName, String context, String version, String provider,String apiTier,String resourceTier, URL endpointUrl)
+            throws APIManagerIntegrationTestException {
+        this.name = apiName;
+        this.context = context;
+        this.version = version;
+        this.provider = provider;
+        this.tiersCollection=apiTier;
+        resourceBeanList = new ArrayList<APIResourceBean>();
+        resourceBeanList.add(new APIResourceBean("GET", "Application & Application User", resourceTier, "/*"));
+        if(endpointUrl != null) {
+            try {
+                this.endpoint = new JSONObject("{\"production_endpoints\":{\"url\":\""
```

Comment:
It might be safer to use a JSON message builder like Gson for generating JSON messages.